python/interpret_text/introspective_rationale/introspective_rationale_explainer.py

```diff
@@ -0,0 +1,322 @@
+from interpret_text.common.utils_classical import plot_local_imp
+from interpret_text.introspective_rationale.introspective_rationale_components import (
+    ClassifierModule,
+    IntrospectionGeneratorModule,
+    ClassifierWrapper,
+)
+from interpret_text.introspective_rationale.introspective_rationale_model import IntrospectiveRationaleModel
+from interpret_text.explanation.explanation import _create_local_explanation
+from interpret_text.common.utils_introspective_rationale import generate_data
+
+import torch
+import numpy as np
+import pandas as pd
+from transformers import BertForSequenceClassification
+
+
+class IntrospectiveRationaleExplainer:
+    """
+    An explainer for training an explainable neural network for natural
+    language processing and generating rationales used by that network.
+    Based on the paper "Rethinking Cooperative Rationalization: Introspective
+    Extraction and Complement Control" by Yu et. al.
+    """
+
+    def __init__(
+        self,
+        args,
+        preprocessor,
+        classifier_type="BERT",
```

Comment:
I don't think classifier type is good here

Reply:
I think we want a base_model input. Where we give Bert and maybe default to an RNN

Reply:
you can leave for now but the api seems off. I would expect, base_model=None, three others. It is mutually exclusive so base_model or all 3 or it fails

Reply:
The intuition here is we should not require our users to install bert by default